INEGI Parser

milmazz opened the pull request:
Los cambios incluyen lo siguiente:
- Agregada una licencia para el proyecto INEGI Parser (_MIT_ como propuesta)
- Agregada sección de autores/contribuyentes
- Sustituida funcionalidad de _scripts bash_ `estatal.sh` y `nacional.sh` por `inegi-ng.py`, el script hace agrega una nueva dependencia: `requests` (actualizado en `requirements.txt`)
- Agregado fichero de configuración general de la aplicación (`source.cfg`), hasta ahora solo incluye _path_ base de los archivos a descargas más las entidades. Posteriormente se incluirá driver de la DB a interrogar, nombre de usuario y password en caso de ser necesario para realizar las conexiones.
- Modificado el formato del `README` (Markdown a rst), lo cual tendrá como ventaja poder incluir toda la información mostrada al "empaquetar" el utilitario vía distutils/setuptools (`setup.py` + `setup.cfg`)
- Agregado `distribute_setup.py` como base a la mejora en el empaquetamiento de la aplicación (próximos pasos)

milmazz committed:
Initial version of the INEGI Parser next gen.

milmazz committed:
Deleted unused files.

milmazz committed:
Deleted non-required dependency.

milmazz committed:
Added intentionally untracked files to ignore.

milmazz committed:
Created 3 entry_points for python console scripts.

Added 3 entry points for console scripts.
Fixed CodeandoMexico/los-scrapers#21

lalo commented:
Sobre la licencia estaria bien ver los pros y cons; otra buena opción es GPL-3 https://tldrlegal.com/license/gnu-general-public-license-v3-(gpl-3) y también revisen la de MIT https://tldrlegal.com/license/mit-license

milmazz commented:
@lalo Sobre la licencia lo podemos conversar, solo que mi recomendación sería no adoptar el esquema GPL, si esperamos introducir nuevos proyectos al conglomerado de `los-scrapers` hay que tener en cuenta que aunque licencias como BSD/MIT son GPL compatibles, la GPL no permite el uso de código licenciado como GPL en código no-GPL, lo cual puede llegar a ser un problema y lo ha sido en varios proyectos, caso _readline_ por ejemplo.

Al menos en mi opinión, apostaría por esquemas de dominio público, como la MIT o
incluso la BSD:

```
Copyright (c) <year>, <copyright holder> All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

    Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
    Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
    Neither the name of the <organization> nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
```

En este caso, la organización sería Codeando México. Tal cual como hacen otras
fundaciones como la _Django Software Foundation_. Si ves también RoR es MIT, así
que también existen casos de éxito bajo licencias permisivas.

Otro punto que veo a favor de las licencias permisivas es que si inicialmente seleccionamos BSD por ejemplo, es totalmente válido re-licenciar bajo GPL si lo quieres, pero lo inverso no es posible, es decir, una vez que hiciste un release GPL no hay vuelta atrás, siempre será GPL, al menos esa versión pues no podrás revocar la licencia.

rafaelcr commented on script/inegi/AUTHORS:
```diff
@@ -0,0 +1,2 @@
+Rafael Cardenas
```
Gracias por dejar la referencia. Puedes agregar mi correo como rafael@rafaelcr.com?

rafaelcr commented on script/inegi/README.rst:
```diff
@@ -1,61 +1,73 @@
+--------------
+
+Para correr el proceso completo, primero hay que ejecutar nacional.sh o
+estatal.sh, estos scripts bajan y descomprimen los archivos de todos los
```
Estas instrucciones deben modificarse para que usen tu nuevo script de python que baja los TSVs en vez de los .sh eliminados.

rafaelcr commented on script/inegi/setup.cfg:
```diff
@@ -0,0 +1,44 @@
+description-file =
+    README.rst
+author = Rafael Cardenas
+author_email = rafaelcr@gmail.com
```
Puedes cambiar este mail a rafael@rafaelcr.com?

rafaelcr commented on script/inegi/distribute_setup.py:
```diff
@@ -0,0 +1,556 @@
+
+This file can also be run as a script to install or upgrade setuptools.
+"""
+import os
```
Me agrada la idea de tener este proceso de setup que instale las dependencias para el scraper, sin embargo creo que este archivo no tiene mucho sentido que sea parte del repo junto a la lógica del scraping. No existe algún repo con este script que podamos agregar como git submodule en su lugar?

milmazz replied:
@rafaelcr El tener embebido `setuptools` puede ser útil en aquellos casos donde el usuario descargue el paquete desde PyPI por ejemplo, en ese caso se baja un `.tar.gz` normalmente, el usuario lo descomprime, ingresa al directorio y con este fichero embebido puede instalarlo (en el caso por supuesto que el usuario no haya instalado previamente _setuptools_). Por supuesto, al tenerlo en PyPI también puedes recurrir a `pip` para instalarlo, pero me pareció útil cubrir el primer caso expuesto.

De todos modos consideraré lo del `submodule` o cambiar un tanto el _approach_ para cubrir la mayor parte de los escenarios.

rafaelcr commented:
Gracias por este PR! Creo que sólo hay que revisar bien lo de separar el setup de la lógica del scraping en general antes de darle merge.

Acerca de la licencia, tú qué opinas @juanpabloe? Creo que sería bueno definir una que sea consistente con todo el objetivo de este repo (es decir, que otros scrapers también puedan compartirla). Si acaso hay algo más de debate sobre esto, creo que sería buena idea mover por lo pronto `LICENSE` a otro PR y continuar el thread allá.

milmazz committed:
Agregado el correo correcto de Rafael Cardenas.

milmazz committed:
Corregido el correo de Rafael Cardenas.

milmazz committed:
Actualizado README, se incluye nuevos scripts.

- Se pone de manifiesto el cambio en la obtención de archivos desde el
INEGI.
- Se incluye proceso de instalación de los scripts.

milmazz committed:
Simplificada la distribución de la app.

milmazz committed:
Ajustado nombre principal al resto de scripts.

milmazz commented:
@lalo @juanpabloe Chicos, ¿qué hace falta para cerrar este PR?, se tomaron en consideración todas las observaciones hechas por @rafaelcr, ¿falta algo más?, pregunto para poder continuar con una nueva mejora, la cual apunta a solventar lo planteado en el _issue_ #22

juanpabloe commented:
@milmazz gracias! de mi lado solamente me gustaría que el license esté en un PR (y conversación) por separado, pues estamos definiendo esto aún algo que sería general para todos los scrapers que se trabajen

milmazz committed:
Eliminada propuesta de licencia MIT.

Se elimina la propuesta de licencia MIT del proyecto INEGIParser en
espera de una respuesta del equipo detrás de Codeando México.

juanpabloe commented:
@milmazz buenísimo! @lalo @rafaelcr cómo ven ? sería bueno mergear este PR

lalo commented:
Se ve todo bien solo quiero correrlo antes para asegurar que funcione la instalación y los cambios. Excelente y gracias!

lalo commented:
Hice una vagrant box con ubuntu para hacer todo el "flow" de instalación desde cero. Me gustaría que el README fuera mas explícito de los requierments pero @milmazz tu no te preocupes de eso. 

Hay un problema después de correr "inegi_get_data" los datos se están instalando en 

/usr/local/lib/python2.7/dist-packages/inegi-0.2.1-py2.7.egg/inegi/datos/

esto es algo que definitivamente no queremos pero así es como lo diseñaste?

milmazz committed:
Agregada licencia AGPLv3.

milmazz committed:
Mejoras en el manejo de argumentos.

- Ahora es posible que el usuario suministre a través del
parámetro -c o --config la localización del fichero de configuración, de lo
contrario, se usa el fichero de configuración que viene por omisión con el
paquete (inegi.cfg).
- Se verifica la existencia de las secciones 'general' y 'entities', así como
el atributo 'base_path' dentro de la sección 'general' en el fichero de
configuración.
- Se agregó un parámetro en donde el usuario deberá indicar el directorio
destino donde los ficheros del INEGI serán depositados, si dicho directorio no
existe será creada la jerarquía necesaria para garantizar su existencia.
- Se actualiza la documentación para reflejar estos cambios.

milmazz commented:
@lalo Cambiado el comportamiento descrito, ahora es posible indicar incluso otro fichero de configuración (siempre y cuando se respete la estructura) así como también es necesario indicar el directorio destino de los ficheros a descargar desde el INEGI. Mayor detalle en los comentarios asociados a los _commits_ previos.

milmazz commented:
@lalo @juanpabloe Chicos, ¿algún comentario respecto a los últimos cambios? :confused:

lalo commented:
Si, aun no lo logro correr exitosamente pero no se si tenga que ver con tu pull request o es algún detalle que ya se tenia. Al correr "inegi_nosql --dbwrite prueba2/*" me sale un error de "IOError: [Errno 2] No such file or directory: 'prueba2/aguas_calientes/aguas_calienNotas.tsv'" y el contenido del folder aguas_calientes es de 

"prueba2/aguas_calientes$ ls
01_aguascalientes_notas.tsv  01_aguascalientes_notasxvalor.tsv  01_aguascalientes_valor.tsv"

Al parecer hubo un cambio de nombre de los archivos.

@milmazz tu lograste correrlo exitosamente? Solo estoy siguiendo los pasos del README hay algo mas que se tenga que configurar?

milmazz commented:
@lalo Esto ya lo indicaba en otro issue en donde decía que el formato
entregado ahora por el INEGI cambio. El script inegi_nosql.py espera
ficheros que ya no entrega el INEGI y no maneja correctamente la excepción.
Ciertamente hay que actualizarlo, ya estaba trabajando en ello pero detuve
la labor hasta aprobar este PR y avanzar paso a paso.

lalo commented:
Ok perfecto, esos son los siguientes pasos entonces.
